TCMSApp/src/client/app/runs/runs-exec/runs-edit.controller.js

```diff
@@ -53,22 +50,33 @@
         }
 
         function runRun() {
-            $http.put('/api/v1/Runs/' + $stateParams.id, {
-                envFull: vm.run.envFull,
-                build: vm.run.build
-            }).success(function (run) {
-                RunsApiService.getTestsOfRun(run._id).query().$promise
-                    .then(function (tests) {
-                        vm.run.tests = tests;
-                        console.log(vm.run);
+            RunsApiService.getTestsOfRun(vm.run._id).query().$promise
+                .then(function (tests) {
+                    vm.run.tests = JSON.parse(JSON.stringify(tests));
```

Comment:
In the tests array there was some $promises(when I have been checking this in the console), and I have eleminated them using JSON methods.